rust/noosphere-core/src/api/client.rs

```diff
@@ -529,3 +567,56 @@ where
         Ok(push_response)
     }
 }
+
+#[cfg(feature = "test-gateway")]
+fn apply_test_header(headers: &mut HeaderMap, forced_host_header: &reqwest::header::HeaderValue) {
+    use reqwest::header::HOST;
+    _ = headers.remove(HOST);
+    headers.insert(HOST, forced_host_header.to_owned());
+}
+
+#[cfg(feature = "test-gateway")]
+fn create_test_header(api_base: &Url, identity: &str) -> Result<reqwest::header::HeaderValue> {
+    let mod_identity = identity
+        .strip_prefix("did:key:")
+        .ok_or_else(|| anyhow!("Could not format HOST header for test-gateway."))?;
+    let domain = api_base.domain().or_else(|| Some("gateway.test")).unwrap();
```

Comment:
Probably want this to be an error, since it is just guessing at this point?

Reply:
domain is `None` in the case of e.g. `127.0.0.1`, where we need some domain name to use to prepend the sphere identity

Reply:
Yeah, that's why I'm saying it should be an error, because you're just guessing that e.g., `gateway.test` is a legible hostname.

Reply:
If the tester explicitly wants that hostname, they can configure it!

Reply:
I'm not sure I follow, this is strictly to test servers at host 127.0.0.1 in tests, if I could configure a host name for local tests this feature wouldn't be needed. `gateway.test` can be anything, but we can't add a subdomain to `127.0.0.1`